constraint/pkg/regorewriter/regorewriter_test.go

```diff
@@ -495,3 +506,256 @@ func TestRegoRewriterErrorCases(t *testing.T) {
 		}
 	}
 }
+
+func TestRegoRewriterRewriteDataRef(t *testing.T) {
+	tests := []struct {
+		name    string
+		prefix  string
+		imports string
+		extern  []string
+		want    string
+	}{
+		{
+			name:    "no rewrite when data ref is an external reference",
+			prefix:  "foo.bar",
+			imports: "data.ext",
+			extern:  []string{"data.ext", "data.inventory"},
+			want:    "data.ext",
+		},
+		{
+			name:    "rewrite for ref is sub of external references",
+			prefix:  "foo.bar",
+			imports: "data.ext",
+			extern:  []string{"data.ext.util"},
+			want:    "data.foo.bar.ext",
+		},
+		{
+			name:    "no rewrite for non data ref",
+			prefix:  "foo.bar",
+			imports: "invalid_data.name",
+			want:    "invalid_data.name",
+		},
+		{
+			name:    "add prefix for data ref",
+			prefix:  "foo.bar",
+			imports: "data.some_ref",
+			want:    "data.foo.bar.some_ref",
+		},
+		{
+			name:    "add prefix same as libs",
+			prefix:  "data.lib",
+			imports: "data.some_ref",
+			want:    "data.data.lib.some_ref",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			externs := []string{"data.inventory"}
+			if tt.extern != nil {
+				externs = tt.extern
+			}
+			r, err := New(NewPackagePrefixer(tt.prefix), []string{"data.lib"}, externs)
```

Comment:
`"data.lib"` is a magic value here. It should either be a test case parameter (if we may want to test what happens when it is a different value) or assigned to a variable with a name that explains what it is.

We'll want to test different values of the libs argument (what if it's unset? does it change the behavior of the package prefixer?), we'll want this as an argument for the test case.

Reply:
Added lib to test parameter, also testing lib not provided cases.